pkg/monitor/agent/cell.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package agent
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/common"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/maps/eventsmap"
+)
+
+// Cell provides the monitor agent, which monitors the cilium events perf event
+// buffer and forwards events to consumers/listeners. It also handles
+// multicasting of other agent events.
+var Cell = cell.Module(
+	"monitor-agent",
+	"Consumes the cilium events map and distributes those and other agent events",
+
+	cell.Provide(new),
+	cell.Config(defaultConfig),
+)
+
+type AgentConfig struct {
+	// EnableMonitor enables the monitor unix domain socket server
+	EnableMonitor bool
+
+	// MonitorQueueSize is the size of the monitor event queue
+	MonitorQueueSize int
+}
+
+var defaultConfig = AgentConfig{
+	EnableMonitor: true,
+}
+
+func (def AgentConfig) Flags(flags *pflag.FlagSet) {
+	flags.Bool("enable-monitor", def.EnableMonitor, "Enable the monitor unix domain socket server")
+
+	// MonitorQueueSizeName is the name of the option MonitorQueueSize
+	flags.Int("monitor-queue-size", 0, "Size of the event queue when reading monitor events")
+}
+
+func new(lc hive.Lifecycle, log logrus.FieldLogger, cfg AgentConfig, _ eventsmap.Map) Agent {
```

Comment:
I'd prefer naming this to `newMonitorAgent` for the sake of "greppability".

Reply:
Since it's not exported the naming doesn't matter much. For exported constructors I'd still recommend just using `New` as per usual Go guidelines. You can always grep for `foo.New`.